app/controllers/reservations_controller.rb

```diff
@@ -304,11 +304,19 @@ def ability_resource
 
   # TODO: you shouldn't be able to edit reservations that have passed or are outside of the cancellation period (check to make sure order has been placed)
   def invalid_for_update?
-    can_edit = @reservation.admin_editable?
-    can_edit &&= @reservation.can_customer_edit? unless current_user.administrator?
     params[:id].to_i != @reservation.id ||
       @reservation.actual_end_at ||
-      !can_edit
+      !editable_by_current_user?
+  end
+
+  def editable_by_current_user?
+    if current_user.administrator? && @reservation.admin_editable?
```

Comment:
I don't think it's worth the effort of changing right now, but in general, we're preferring switching things like `current_user.administrator?` to a check on abilities.